File: app/src/main/java/me/zzy/redot/room/entity/Board.java
package me.zzy.redot.room.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

/**
 * @author ZZY
 * 12/2/20.
 */
@Entity
public class Board {
    @PrimaryKey(autoGenerate = true)
    public int boardId;

    @ColumnInfo(name = "text")
    public String text;

    @ColumnInfo(name = "text_size")
    public float textSize;

    @ColumnInfo(name = "text_color")
    public int text_color;

    @ColumnInfo(name = "background_color")
    public int background_color;

    @ColumnInfo(name = "speed")
    public int speed;

    @ColumnInfo(name = "is_horizontal")
    public boolean isHorizontal;

    @ColumnInfo(name = "clickable")
    public boolean clickable;

    @ColumnInfo(name = "is_scroll_forever")
    public boolean is_scroll_forever;

    @ColumnInfo(name = "times")
    public int times;

}

File: app/src/main/java/me/zzy/redot/room/entity/Board.kt
package me.zzy.redot.room.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * @author ZZY
 * 12/2/20.
 */
@Entity
data class Board(
        @PrimaryKey(autoGenerate = true)
        var boardId: Int = 0,

        @ColumnInfo(name = "text")
        var text: String? = null,

        @ColumnInfo(name = "text_size")
        var textSize: Float = 0f,

        @ColumnInfo(name = "text_color")
        var textColor: Int = 0,

        @ColumnInfo(name = "background_color")
        var backgroundColor: Int = 0,

        @ColumnInfo(name = "speed")
        var speed: Int = 0,

        @ColumnInfo(name = "is_horizontal")
        var isHorizontal: Boolean = false,

        @ColumnInfo(name = "clickable")
        var clickable: Boolean = false,

        @ColumnInfo(name = "is_scroll_forever")
        var isScroll: Boolean = false,

)